node/src/actors/chain_manager/actor.rs

```diff
@@ -254,6 +256,17 @@ impl ChainManager {
 
                 act.chain_state = chain_state;
 
+                // Update possible new WIP information
+                let (new_wip_epoch, old_wips) = act.chain_state.tapi_engine.initialize_wip_information();
+                let last_consolidated_epoch = act.get_chain_beacon().checkpoint;
+                if new_wip_epoch < last_consolidated_epoch {
+                    ChainManager::from_registry().do_send(UpdateWipVotes{
+                        init: new_wip_epoch,
+                        end: last_consolidated_epoch,
+                        old_wips
+                    });
```

Comment:
I think it's important that this message is completed before the chain manager tries to do anything else, so use `.send().wait(ctx)` instead of `do_send()`.